pyro/infer/enum.py

```diff
@@ -32,14 +42,66 @@ def iter_discrete_traces(graph_type, fn, *args, **kwargs):
     queue.put(Trace())
     q_fn = poutine.queue(fn, queue=queue)
     while not queue.empty():
-        q_fn = poutine.queue(fn, queue=queue)
-        full_trace = poutine.trace(
-            q_fn, graph_type=graph_type).get_trace(*args, **kwargs)
+        q_fn = poutine.queue(fn, queue=queue, escape_fn=_iter_discrete_escape)
```

Comment:
Should this be outside the while loop replacing `q_fn` above?

Reply:
That looks right to me. Done.